w3lib/url.py

```diff
@@ -56,8 +58,8 @@ def safe_url_string(url, encoding='utf8', path_encoding='utf8'):
     #     encoded with the supplied encoding (or UTF8 by default)
     #   - if the supplied (or default) encoding chokes,
     #     percent-encode offending bytes
-    parts = urlsplit(to_unicode(url, encoding=encoding,
-                                errors='percentencode'))
+    decoded = to_unicode(url, encoding=encoding, errors='percentencode')
+    parts = urlsplit(remove_tab_newline.sub('', decoded))
```

Comment:
I think this newline/tab removing should be documented in safe_url_string docstring - something about workaround to make parser more compatible with https://url.spec.whatwg.org/#url-parsing, maybe "as per https://url.spec.whatwg.org/#url-parsing, tabs and newlines are removed"